esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.h

```diff
@@ -0,0 +1,2145 @@
+     * traditional text buffer, which remains available.
+     */
+    virtual void setInput(const IXmlFragment& fragment) = 0;
+
+    /**
+     * Inserts a fragment's content into the parser while parsing existing input. The parser will
+     * suspend processing of the current fragment to parse the new fragment. Upon completion of the
+     * new fragment, the parser will continue processing the original fragment.
+     */
+    virtual void injectFragment(const IXmlFragment& fragment) = 0;
+};
+
+
+/**
+ * Instruct the parser to retain a fragment's parent element. When an element identifies a fragment
+ * the parser's default behavior is to replace the original element, in its entirety, with the
+ * fragment. This instructs the parser to retain the fragment identifying element and inject the
+ * fragment markup as descendents of that element.
+ */
+#define FXPP_RETAIN_FRAGMENT_PARENT   XmlFragmentRules(0x01)
+
+/**
+ * Make all namespaces in effect at the time of injection available to the injected content. Each
+ * namespace URI and prefix is defined in the fragment, eliminating the need to repeat definitions
+ * in each fragment.
+ * 
+ * By default, namespaces are not propagated. Any fragment requiring propagation must request it
+ * when constructed. Certain `IXmlFragmentGroup` implementations do use this flag to ensure that
+ * group members created by the group can inherit namespaces defined for the group.
+ */
+#define FXPP_PROPAGATE_NAMESPACES     XmlFragmentRules(0x04)
+
+/**
+ * Instruct the parser to require a non-empty unique identifier for a fragment. The identifier is
+ * used to detect circular references between fragments. Identifiers are required for all fragments
+ * with the exception of the initial parser input and embedded fragments.
+ *
+ * Some identifiers are generated automatically by fragment implementations, while others must be
+ * assigned manually. Fragments with manually assigned values must assign the same identifier to
+ * every occurrence of the same content. For example, two references to the same external fragment
+ * may produce two separate fragment instances but both instances must specify the same identifier.
+ */
+#define FXPP_REQUIRE_FRAGMENT_UID     XmlInternalFragmentRules(0x0100)
+
+/**
+ * Instruct the parser to enforce the injection restriction requiring an in-progress element.
+ */
+#define FXPP_REQUIRE_CURRENT_ELEMENT  XmlInternalFragmentRules(0x0200)
+
+/**
+ * Reserved element names used internally by the parser.
+ */
+namespace reserved
+{
+    constexpr static const char* SYNTHESIZED_ROOT = "synthesized_root";
+    constexpr static const char* SYNTHESIZED_GROUP = "synthesized_group";
+}
+
+/**
+ * `IXmlFragment` implementation that assumes the character pointers given in its constructor will
+ * remain valid for the life of the instance.
+ */
+class CSimpleXmlFragment : public CInterfaceOf<IXmlFragment>
```

Comment:
I think this class would lend itself to unsafe behavior in real world implementations.  It ok for testing and internal use, but maybe it should be local static implementations of this type of class in those cpp files.  Don't want to make people think this class is safe to use.  By unsafe I mean that we don't know how long the black box will be referencing these pointers and could for example cause shut down issues.  Another class that did reference counting around the content would be better.